packages/glimmer-compiler/lib/compiler.ts

```diff
@@ -19,7 +25,15 @@ export type TemplateSpec = string;
  * @return {TemplateSpec} A template spec string
  */
 export function compileSpec(string: string, options: any): TemplateSpec {
+  let meta: BlockMeta = {};
+  if (typeof options === 'object') {
+    meta = Object.assign(meta, options.meta);
+  }
   let ast = preprocess(string, options);
   let program = TemplateCompiler.compile(options, ast);
-  return JSON.stringify(program);
+  let templateModule: TemplateModule = {
+    meta,
+    template: JSON.stringify(program)
+  };
+  return JSON.stringify(templateModule);
```

Comment:
I think my pseudo code was confusing because I was explaining precompile, which currently results in a stringify(stringify(templateSpec)) but doing this in `compileSpec` results in 3 stringifies